cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java

```diff
@@ -193,6 +197,10 @@ public static void afterClass() throws Exception {
     txManager.stopAndWait();
   }
 
+  protected UsageHandler getUsageHandler() throws IOException, DatasetManagementException {
```

Comment:
Actually, the issue was that some of the operations on `UsageDataset` wasn't in a transaction, so the operations weren't getting flushed.